### PR TITLE
[DOCS] Adds Connecting section to docs

### PR DESCRIPTION
**Pull request by szabosteve**
## Overview

This PR adds a `Connecting` section to the documentation.

### Preview

[Connecting](https://elasticsearch-net_6183.docs-preview.app.elstc.co/guide/en/elasticsearch/client/net-api/master/connecting.html)

### CODE REVIEW

**Comment by szabosteve** [comment]
@elasticmachine test this please

File: docs/connecting.asciidoc [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+
+[source,csharp]
+----
+using Elastic.Clients.Elasticsearch;
+using Elastic.Transport;
+
+var client = new ElasticsearchClient("<CLOUD_ID>", new ApiKey("<API_KEY>"));
+----
+
+Replace the placeholder string values above with your cloud ID and the API key
```
Comment (stevejgordon) [inline docs/connecting.asciidoc:45]: I think we should make this one of those numbered code comments (not sure what we call them), with the blue numbered circle as this directly references the line of code it relates to?

File: docs/connecting.asciidoc [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+Single node configuration is best suited to connections to a multi-node cluster 
+running behind a load balancer or reverse proxy, which is exposed via a single 
+URL. It may also be convenient to use a single node during local application 
+development. If the URL represents a single {es} node, be aware that this offers 
+no resiliency should the server be unreachable or unresponsive.
+
+By default, with security features like authentication and TLS enabled. When you 
+start {es} for the first time, TLS is configured automatically for the HTTP 
+layer. A CA certificate is generated and stored on disk that is used to sign the 
+certificates for the HTTP layer of the {es} cluster.
```
Comment (stevejgordon) [inline docs/connecting.asciidoc:62]:
```suggestion
By default, security features such as authentication and TLS are enabled on {es} clusters. 
When you start {es} for the first time, TLS is configured automatically for the HTTP 
layer. A CA certificate is generated and stored on disk which is used to sign the 
certificates for the HTTP layer of the {es} cluster.
```

File: docs/connecting.asciidoc [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+
+In order for the client to establish a connection with the cluster over HTTPS, 
+the CA certificate must be trusted by the client application. There are several 
+mechanisms for working with certificates, but the simplest choice is to use the
```
Comment (stevejgordon) [inline docs/connecting.asciidoc:66]: We need a link here:

```suggestion
mechanisms for <<working-with-certificates, working with certificates>>, but the simplest choice is to use the 
```

**Review by stevejgordon — CHANGES_REQUESTED** [review]
A few minor changes but otherwise LGTM! Thanks!

**Review by stevejgordon — APPROVED** [review]
LGTM! Thanks again!

**Pull request merged**